### DISCUSSION TITLE
Does hide_by_pattern support paths?

### DISCUSSION THREAD

**Discussion by johnf** [nvim-neo-tree/neo-tree.nvim#464] [Q&A]
In nvim-tree I could hide fill paths.
My use case is I have a bunch of directories one per lambda function
They each have symlinks to things like tsconfig.json and a dist directory for building.

I've tried
```lua
hide_by_pattern = {
        'src/lambda/*/tsconfig.json',
        'src/lambda/**/tsconfig.json',
        '*src/lambda/*/tsconfig.json',
        'src/lambda/favours/acceptFavour/tsconfig.json',
}
```

all with no luck

**Comment by cseickel** [comment]
It might just be that the word `pattern` is misspelled. I would expect this to work:

```lua
    require("neo-tree").setup({
      filesystem = {
        filtered_items = {
          hide_by_pattern = { 
            '*/src/lambda/*/tsconfig.json',
          }
        }
      }
    })
```

**Reply by johnf** [reply]
That was a typo above. (which I've edited).

I added some debugging in the code

```lua
    if f.hide_by_pattern then
      for _, p in ipairs(f.hide_by_pattern) do
        log.error(name)
        if string.find(name, p) then
          item.filtered_by = item.filtered_by or {}
          item.filtered_by.pattern = true
        end
      end
    end
```
It's printing out just the filename, so I don't think it can match the full path.

If I change ```name``` to ```path``` it works perfectly.

I'm happy to submit a PR for this, but I have a feeling it's a breaking change and might cause unexpected behaviour for folks with existing patterns

**Reply by cseickel** [reply]
I think a good solution would be to test if there is a path separator in the pattern, and if there is, use the `path` property, else use the `name`. This way it will support your use case without affecting existing patterns of other users.

I'll happily merge a PR that does this.

**Reply by johnf** [reply]
PR #468 , let me know if you need any changes
